nearuplib/localnet.py

```diff
@@ -40,23 +44,34 @@ def run(args):
     mkdir(LOGS_FOLDER)
 
     # Spawn network
-    pid_fd = open(NODE_PID, 'w')
-    for i in range(0, args.num_nodes):
-        proc = run_binary(args.binary_path, join(args.home, f'node{i}'), 'run', verbose=args.verbose,
-                          boot_nodes=f'{pk}@127.0.0.1:24567' if i > 0 else None, output=join(LOGS_FOLDER, f'node{i}'))
-        proc_name = proc_name_from_pid(proc.pid)
-        print(proc.pid, "|", proc_name, "|", 'localnet', file=pid_fd)
-    pid_fd.close()
+    if args.docker_image:
+        for i in range(0, args.num_nodes):
+            run_docker(args.docker_image, join(args.home, f'node{i}'), boot_nodes=f'{pk}@127.0.0.1:24567' if i > 0 else None,
+                       verbose=args.verbose, container_name=f'nearcore{i}', host_network=True)
+    else:
+        pid_fd = open(NODE_PID, 'w')
+        for i in range(0, args.num_nodes):
+            proc = run_binary(args.binary_path, join(args.home, f'node{i}'), 'run', verbose=args.verbose,
+                              boot_nodes=f'{pk}@127.0.0.1:24567' if i > 0 else None, output=join(LOGS_FOLDER, f'node{i}'))
+            proc_name = proc_name_from_pid(proc.pid)
+            print(proc.pid, "|", proc_name, "|", 'localnet', file=pid_fd)
+        pid_fd.close()
 
     print("Local network was spawned successfully.")
-    print(f"Check logs at: {LOGS_FOLDER}")
+    if args.docker_image:
```

Comment:
Would be great if `nearup logs` was printing logs independently on how it was launched.

Reply:
will address in a separate PR